common/common_target.mk

```diff
@@ -254,6 +272,11 @@ ifeq ($(TARGET_NATIVE_COMPILED), true)
 		NATIVE_LIBRARY_PATHS+=$(LIBRARY_PATHS)
 		NATIVE_LIBRARY_PATHS+=$(LINUX_LIBRARY_PATHS)
 	endif
+	ifeq ($(OS), osx)
+		NATIVE_LIBRARY_PATHS+=$(LIBRARY_PATHS)
+		NATIVE_LIBRARY_PATHS+=$(OSX_LIBRARY_PATHS)
+		NATIVE_LDFLAGS+=$(OSX_LDFLAGS)
```

Comment:
we need a better place to put this. commit this for now, and I will add a section for the cc, cxx and ld flags to be OS specific as well.